src/client/app/effects/snippet.ts

```
@@ -187,6 +188,48 @@ export class SnippetEffects {
         .map((updatedSnippet) => new Snippet.SaveAction(updatedSnippet))
         .catch(exception => Observable.of(new UI.ReportErrorAction(Strings().snippetUpdateError, exception)));
 
+    @Effect({ dispatch: false })
+    openInPlayground$: Observable<Action> = this.actions$
+        .ofType(Snippet.SnippetActionTypes.OPEN_IN_PLAYGROUND)
+        .map(action => action.payload)
+        .map(payload => {
+            if (Utilities.platform === PlatformType.IOS) {
+                AI.trackEvent('Unsupported open in playground');
+                return;
+            }
+            let { type, id, isDownload } = payload;
+            let handler, extension;
+            switch (environment.current.host.toUpperCase()) {
+                case HostType.EXCEL:
+                    handler = 'ms-excel:ofe|u|';
+                    extension = '.xlsx';
+                    break;
+                case HostType.WORD:
+                    handler = 'ms-word:ofe|u|';
+                    extension = '.docx';
+                    break;
+                case HostType.POWERPOINT:
+                    case HostType.WORD:
+                    handler = 'ms-powerpoint:ofe|u|';
+                    extension = '.pptx';
+                    break;
+                default:
+                    return;
```

Please throw an error in `default` instead of a silent return.